sbt-ash/src/main/scala/org/virtuslab/ash/DumpEventSchema.scala

```diff
@@ -26,14 +26,14 @@ object DumpEventSchema {
       writer <-
         outputFile.createIfNotExists(createParents = true).clear().newFileOutputStream().printWriter().autoClosed
     } writer.print(yamlPrinter.pretty(json))
+    outputFile
   }
 
-  def dumpEventSchemaTask(key: TaskKey[Unit]): Def.Setting[_] =
+  def dumpEventSchemaTask(key: TaskKey[File]): Def.Setting[_] =
     key := {
-      val c = (Compile / compile).value
-      c.readCompilations()
+      val _ = (Compile / compile).value
       DumpEventSchema(
-        (key / dumpEventSchemaOutputFile).value.toScala,
-        (key / dumpEventSchemaCompilerPluginOutputFile).value.toScala)
+        (key / ashDumpPersistenceSchemaOutputFile).value.toScala,
+        ashCompilerPluginCacheDirectory.value.toScala / "dump-persistent-schema-cache").toJava
```

Comment:
Yeah, but regardless of the renaming (`File`/`JFile`/`SFile`)... why is Java representation of the file needed anywhere in sbt in the first place? 🤔

Reply:
I ma no sure if I understand the question. Partial results are stored in a file, and the schema dump is also a file, so naturally we need some abstraction over filesystem.

Reply:
Yes, sure... but the question is mostly myself being curious why this abstraction happens to be Java-specific (?) at certain levels and Scala-specific at some other ones 🤔

Reply:
FQCN of Java files is `java.io.File` and the scala ones is `better.files.File`. Sbt uses Java files and I use the one's from library `better-files` because they are... better.

Reply:
Awww crap, `better-files`, that's what I was missing 🤦🏻 okay, thx